sdk/containerregistry/azure-containerregistry/tests/testcase.py

```diff
@@ -162,7 +162,11 @@ def get_credential(self):
         return FakeTokenCredential()
 
     def create_registry_client(self, endpoint, **kwargs):
-        return ContainerRegistryClient(endpoint=endpoint, credential=self.get_credential(), **kwargs)
+        if endpoint.endswith("azurecr.io"):
+            return ContainerRegistryClient(endpoint=endpoint, credential=self.get_credential(), **kwargs)
+        if endpoint.endswith("azurecr.cn"):
+            return ContainerRegistryClient(endpoint=endpoint, credential=self.get_credential(), authorization_scope="https://portal.azure.cn", **kwargs)
```

Comment:
I think it would be better for us to put these portal endpoints in the subscription configuration, that way you don't have to do any switching in the code (assuming we can pass authorization_scope for public as well).